[BB-4023] Create MicroSite Configuration on different services (#1)
open-craft/microsite-generator: README.md

# microsite-generator

Generates MicroSites for an edX installation.

# Usage

- Copy `config/sample.yaml` to `config/config.yaml`.
- Check `Makefile` for available commands.
```
make help

help:     Show help.

run:  Generate microsite configuration for all services
run-lms:  Generate microsite configuration in LMS
run-discovery:  Generate microsite configuration in Discovery
run-ecommerce:   Generate microsite configuration in eCommerce

dev.run-lms:     Devstack - Generate microsite configuration in LMS
dev.run-discovery:   Devstack - Generate microsite configuration in Discovery
dev.run-ecommerce:   Devstack - Generate microsite configuration in eCommerce
```

> # Structure of config file

Reviewer: A little more intro on how to set up this project, will go a long way since we might have to tell the client on how to achieve this.

Author: Added more documentation on how to install and use it.

`config/config.yaml` file contains following features -

- `main_domain`: The base domain. Each microsite will be set as subdomain of this domain.
- `site_for_each_organization`: Set this to `true` if each organization will have a microsite. Then the script will automatically copy `organizations` as `microsites`.
- `organizations`: A dictionary containing organization related info
    ```yaml
    organizations:
        short_code:
            name: Name of the organization
    ```
- `microsites`: A dictionary containing microsite related info -
    ```yaml
    microsites:
        sub_domain:
            name: Name of the microsite
            context_overrides: {}
            overrides: {}
    ```

### Overrides

> It might be necessary to override some value of some field in microsite configuration. For that we have following options -

Reviewer: ```suggestion
To override values of fields in microsite configuration, we have the following options 
```

Author: Thanks for simplifying this. Updated.
